Code/Framework/AzCore/AzCore/Name/Name.h

```diff
@@ -45,22 +54,45 @@ namespace AZ
         Name();
         Name(const Name& name);
         Name(Name&& name);
+        template<size_t N>
+        explicit Name(const char (&literalString)[N])
```

Comment:
Technically this doesn't work if the user declare a char buffer and then supplied ti to this function.
i.e
```
const char buffer[256] = "Hello";
Name helloName(buffer);
```

Reply:
Yep, there's two edge cases:
1. Calling this a constructor with a non-literal buffer - it'll still work if the Name dictionary is initialized, otherwise it'll fail. I think this is a pretty low-likelihood use-case, but I can document it.
2. The converse case where you do want to make a Name literal but don't have a `char[]` - you can use `Name::FromStringLiteral` for that - I'm debating if it's worth keeping this around as public API, but it does let you do e.g.
```
static constexpr AZStd::string_view FooText("Foo");
static constexpr AZ::Name Foo = AZ::Name::FromStringLiteral(FooText);
```

Reply:
I think I'd be in favor of removing the string literal constructor and just require FromStringLiteral() be used for any static initialization. That will remove any ambiguity or confusion of which to use when.
Code reviewers already have to be on the lookout for static Names in general, I think it's not worse for code reviewers to be on the lookout for static Names that fail to use FromStringLiteral().

Reply:
I've removed the templated constructor in favor of enforcing `FromStringLiteral` for static names